flink-ml-benchmark/src/main/java/org/apache/flink/ml/benchmark/datagenerator/common/LabeledPointWithWeightGenerator.java

```diff
@@ -0,0 +1,202 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.benchmark.datagenerator.common;
+
+import org.apache.flink.api.common.functions.RichMapFunction;
+import org.apache.flink.api.common.typeinfo.BasicTypeInfo;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.ml.benchmark.datagenerator.InputDataGenerator;
+import org.apache.flink.ml.benchmark.datagenerator.param.HasVectorDim;
+import org.apache.flink.ml.common.datastream.TableUtils;
+import org.apache.flink.ml.common.param.HasFeaturesCol;
+import org.apache.flink.ml.common.param.HasLabelCol;
+import org.apache.flink.ml.common.param.HasWeightCol;
+import org.apache.flink.ml.linalg.Vectors;
+import org.apache.flink.ml.linalg.typeinfo.DenseVectorTypeInfo;
+import org.apache.flink.ml.param.IntParam;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+import org.apache.flink.ml.util.ParamUtils;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.NumberSequenceIterator;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+/** A DataGenerator which creates a table of features, label and weight. */
+public class LabeledPointWithWeightGenerator
+        implements InputDataGenerator<LabeledPointWithWeightGenerator>,
+                HasFeaturesCol<LabeledPointWithWeightGenerator>,
+                HasWeightCol<LabeledPointWithWeightGenerator>,
+                HasLabelCol<LabeledPointWithWeightGenerator>,
+                HasVectorDim<LabeledPointWithWeightGenerator> {
+    private final Map<Param<?>, Object> paramMap = new HashMap<>();
+
+    public static final Param<Integer> LABEL_ARITY =
+            new IntParam(
+                    "labelArity",
+                    "Arity of label. "
+                            + "If set to positive value, label would be a positive integer between 0 and arity - 1. "
+                            + "If set to zero, label would be continuous double value.",
+                    2,
+                    ParamValidators.gtEq(0));
+
+    public int getLabelArity() {
+        return get(LABEL_ARITY);
+    }
+
+    public LabeledPointWithWeightGenerator setLabelArity(int value) {
+        return set(LABEL_ARITY, value);
+    }
+
+    public static final Param<Integer> FEATURES_ARITY =
+            new IntParam(
+                    "featuresArity",
```

Comment:
`features` is a dense vector. `featuresArity` might be confused with the rank of this dense vector.

How about renaming this parameter as `featureArity`, since it is the arity of each individual feature value?